htlcswitch/failure.go

```diff
@@ -247,7 +247,7 @@ var _ ErrorEncrypter = (*SphinxErrorEncrypter)(nil)
 // SphinxErrorDecrypter wraps the sphinx data SphinxErrorDecrypter and maps the
 // returned errors to concrete lnwire.FailureMessage instances.
 type SphinxErrorDecrypter struct {
```

Comment:
Could just be an interface now?

Reply:
You mean `type SphinxErrorDecrypter interface`? But what is implementing this interface then? `SphinxErrorDecrypter` offers a method `DecryptError` that depends on an underlying decryptor function `Decrypt`

Reply:
The `OnionErrorDecrypter` would implement this interface. A struct with a single field that is a function closure aspires to be an interface.

Reply:
Actually, we already have `ErrorDecrypter `, why is that insufficient?

Reply:
`ErrorDecrypter` is a higher level interface that returns a `ForwardingError`.

Converted function closure to interface.